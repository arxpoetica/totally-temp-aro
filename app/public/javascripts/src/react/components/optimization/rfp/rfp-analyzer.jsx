import React, { Component } from 'react'
// import { PropTypes } from 'prop-types'
import reduxStore from '../../../../redux-store'
import wrapComponentWithProvider from '../../../common/provider-wrapped-component'
import RfpOptions from './rfp-options.jsx'
import RfpTargets from './rfp-targets.jsx'
import RfpStatusTypes from './constants'

export class RfpAnalyzer extends Component {
  render () {
    var displayOnly = this.props.status !== RfpStatusTypes.UNINITIALIZED
    return <div>
      <RfpOptions initialValues={this.props.rfpOptions} enableReinitialize displayOnly={displayOnly} />
      <RfpTargets displayOnly={displayOnly} />
    </div>
  }
}

RfpAnalyzer.propTypes = {
}

const mapStateToProps = (state) => ({
  rfpOptions: state.optimization.rfp.options,
  status: state.optimization.rfp.status
})

const mapDispatchToProps = dispatch => ({
})

const RfpAnalyzerComponent = wrapComponentWithProvider(reduxStore, RfpAnalyzer, mapStateToProps, mapDispatchToProps)
export default RfpAnalyzerComponent
