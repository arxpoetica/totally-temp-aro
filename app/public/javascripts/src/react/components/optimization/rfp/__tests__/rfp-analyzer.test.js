/* global test expect jest */
import React from 'react'
import { shallow } from 'enzyme'
import { RfpAnalyzer } from '../rfp-analyzer'
import RfpStatusTypes from '../constants'

// -----------------------------------------------------------------------------
test('Default component render', () => {
  const mockInitialize = jest.fn()
  const mockClearState = jest.fn()
  const component = shallow(
    <RfpAnalyzer initialize={mockInitialize}
      clearState={mockClearState}
      status={RfpStatusTypes.UNINITIALIZED}
    />
  )
  expect(component).toMatchSnapshot()
  expect(mockInitialize.mock.calls.length).toBe(1)
  component.unmount()
  expect(mockClearState.mock.calls.length).toBe(1)
})

// -----------------------------------------------------------------------------
test('Rfp running status', () => {
  const mockInitialize = jest.fn()
  const mockClearState = jest.fn()
  const component = shallow(
    <RfpAnalyzer initialize={mockInitialize}
      clearState={mockClearState}
      status={RfpStatusTypes.RUNNING}
    />
  )
  expect(component).toMatchSnapshot()
})

// -----------------------------------------------------------------------------
test('Rfp finished status', () => {
  const mockInitialize = jest.fn()
  const mockClearState = jest.fn()
  const component = shallow(
    <RfpAnalyzer initialize={mockInitialize}
      clearState={mockClearState}
      status={RfpStatusTypes.FINISHED}
    />
  )
  expect(component).toMatchSnapshot()
})
