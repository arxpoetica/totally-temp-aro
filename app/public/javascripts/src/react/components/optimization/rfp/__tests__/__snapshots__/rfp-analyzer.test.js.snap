// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default component render 1`] = `
<div>
  <ReduxForm
    destroyOnUnmount={true}
    enableReinitialize={true}
    forceUnregisterOnUnmount={false}
    form="rfpOptions"
    getFormState={[Function]}
    keepDirtyOnReinitialize={false}
    persistentSubmitErrors={false}
    pure={true}
    shouldAsyncValidate={[Function]}
    shouldError={[Function]}
    shouldValidate={[Function]}
    shouldWarn={[Function]}
    submitAsSideEffect={false}
    touchOnBlur={true}
    touchOnChange={false}
    updateUnregisteredFields={false}
  />
  <Component />
</div>
`;

exports[`Rfp finished status 1`] = `
<div>
  <ReduxForm
    destroyOnUnmount={true}
    enableReinitialize={true}
    forceUnregisterOnUnmount={false}
    form="rfpOptions"
    getFormState={[Function]}
    keepDirtyOnReinitialize={false}
    persistentSubmitErrors={false}
    pure={true}
    shouldAsyncValidate={[Function]}
    shouldError={[Function]}
    shouldValidate={[Function]}
    shouldWarn={[Function]}
    submitAsSideEffect={false}
    touchOnBlur={true}
    touchOnChange={false}
    updateUnregisteredFields={false}
  />
  <Component />
  <div
    className="disable-sibling-controls"
  />
</div>
`;

exports[`Rfp running status 1`] = `
<div>
  <ReduxForm
    destroyOnUnmount={true}
    enableReinitialize={true}
    forceUnregisterOnUnmount={false}
    form="rfpOptions"
    getFormState={[Function]}
    keepDirtyOnReinitialize={false}
    persistentSubmitErrors={false}
    pure={true}
    shouldAsyncValidate={[Function]}
    shouldError={[Function]}
    shouldValidate={[Function]}
    shouldWarn={[Function]}
    submitAsSideEffect={false}
    touchOnBlur={true}
    touchOnChange={false}
    updateUnregisteredFields={false}
  />
  <Component />
  <div
    className="disable-sibling-controls"
  />
</div>
`;
